modules/network/src/main/java/org/apache/ignite/internal/network/serialization/DefaultClassDescriptors.java

```diff
@@ -0,0 +1,140 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network.serialization;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.UUID;
+import org.apache.ignite.lang.IgniteUuid;
+
+/**
+ * Class that holds default types and creates descriptors for them.
+ */
+public class DefaultClassDescriptors {
+    /**
+     * Creates a descriptor for one of the default types.
+     *
+     * @param type Default type.
+     * @return Descriptor.
+     */
+    public static ClassDescriptor createDefaultDescriptor(DefaultType type) {
+        Class<?> clazz = type.clazz;
+        int descriptorId = type.descriptorId;
+        return new ClassDescriptor(
+            ClassDescriptorParser.className(clazz),
+            descriptorId,
+            Collections.emptyList(),
+            SerializationType.DEFAULT,
+            ClassDescriptorParser.isFinal(clazz)
+        );
+    }
+
+    /**
+     * Default types.
+     */
+    public enum DefaultType {
+        BYTE_P(0, byte.class),
+        BYTE(1, Byte.class),
+        SHORT_P(2, short.class),
+        SHORT(3, Short.class),
+        INT_P(4, int.class),
+        INT(5, Integer.class),
+        FLOAT_P(6, float.class),
+        FLOAT(7, Float.class),
+        LONG_P(8, long.class),
+        LONG(9, Long.class),
+        DOUBLE_P(10, double.class),
+        DOUBLE(11, Double.class),
+        BOOLEAN_P(12, boolean.class),
+        BOOLEAN(13, Boolean.class),
+        CHAR_P(14, char.class),
+        CHAR(16, Character.class),
+        OBJECT(17, Object.class),
+        STRING(18, String.class),
+        UUID(19, UUID.class),
+        IGNITE_UUID(20, IgniteUuid.class),
+        DATE(21, Date.class),
+        BYTE_ARRAY(22, byte[].class),
+        SHORT_ARRAY(23, short[].class),
+        INT_ARRAY(24, int[].class),
+        FLOAT_ARRAY(25, float[].class),
+        LONG_ARRAY(26, long[].class),
+        DOUBLE_ARRAY(27, double[].class),
+        BOOLEAN_ARRAY(28, boolean[].class),
+        CHAR_ARRAY(29, char[].class),
+        OBJECT_ARRAY(30, Object[].class),
+        STRING_ARRAY(31, String[].class),
+        DECIMAL(32, BigDecimal.class),
+        DECIMAL_ARRAY(33, BigDecimal[].class),
+        ENUM(34, Enum.class),
+        ENUM_ARRAY(35, Enum[].class),
+        ARRAY_LIST(36, ArrayList.class),
```

Comment:
Well, I suppose a user expects that a linked list will still be a linked list on the other side. Anyway, that's how we do it in ignite-2

Reply:
This makes sense. But, on the other hand, there will always be omittions. For example, there is no `CopyOnWriteArrayList`, so in some cases the user will get exactly what they had before serialization, but in other cases something different will be obtained.

This will need a specific and visible mention in the documentation, I guess... :)

Reply:
Of course, this should be documented as hell